tests/test_tk_config_update.py

```diff
@@ -0,0 +1,217 @@
+    """
+    Ensure the tool enumerates files correctly.
+    """
+    files_found = [
+        # strip the root of the path and flip the separators so we can
+        # compare with the constant at the bottom of the file.
+        path.replace(cloned_config + os.path.sep, "").replace(os.path.sep, "/")
+        for path in tk_config_update.enumerate_yaml_files(cloned_config)
+    ]
+    assert set(files_found) == expected_config_files
+
+
+@pytest.mark.parametrize(
+    "descriptor, expected_result",
+    [
+        ({"something": "something"}, False),
+        ({"type": "shotgun", "name": "something", "version": "231"}, False),
+        ({"type": "app_store", "name": "something", "version": "231"}, True),
+    ],
+)
+def test_is_app_store_descriptor(descriptor, expected_result):
+    """
+    Ensure appstore descriptor detection works.
+    """
+    assert tk_config_update.is_app_store_descriptor(descriptor) is expected_result
+
+
+@pytest.mark.parametrize(
+    "modified_file,path_to_descriptor,bundle,expected_version",
+    [
+        (os.path.join("core", "core_api.yml"), ["location"], "tk-core", "v0.18.0"),
+        (
+            "env/includes/common/frameworks.yml",
+            ["frameworks", "tk-framework-shotgunutils_v5.x.x", "location"],
+            "tk-framework-shotgunutils",
+            "v5.0.0",
+        ),
+        (
+            "env/includes/common/frameworks.yml",
+            ["frameworks", "tk-framework-shotgunutils_v4.x.x", "location"],
+            "tk-framework-shotgunutils",
+            "v4.0.0",
+        ),
+        (
+            "env/includes/common/apps.yml",
+            ["common.apps.tk-multi-publish2.location"],
+            "tk-multi-publish2",
+            "v10.0.0",
+        ),
+        (
+            "env/includes/common/engines.yml",
+            ["common.engines.tk-3dsmax.location"],
+            "tk-3dsmax",
+            "v11.0.0",
+        ),
+    ],
+)
+def test_update_config(
+    cloned_config,
+    test_config,
+    modified_file,
+    path_to_descriptor,
+    bundle,
+    expected_version,
+):
+    """
+    Ensure the update updates the right file and puts the right content in it.
+    """
+    # Make the path valid of the current os.
+    modified_file = modified_file.replace("/", os.path.sep)
+    updated_files = list(
+        tk_config_update.update_files(test_config, bundle, expected_version)
+    )
+    assert updated_files == [os.path.join(test_config, modified_file)]
+
+    with open(os.path.join(cloned_config, modified_file), "rt") as fh:
+        expected_cfg = yaml.load(fh, Loader=yaml.Loader)
+
+    with open(os.path.join(test_config, modified_file), "rt") as fh:
+        test_config = yaml.load(fh, Loader=yaml.Loader)
+
+    value = expected_cfg
+    for key in path_to_descriptor:
+        value = value[key]
+
+    value["version"] = expected_version
+
+    assert expected_cfg == test_config
+
+
+expected_config_files = set(
+    [
+        "env/includes/desktop2/site.yml",
```

Comment:
Is this test going to need updating each time we add a new engine?

Reply:
The fixture clones tk-config-basic at version v1.3.0, so no. I'll add a comment.